src/PrestaShopBundle/EventListener/AssignSmartyVariablesListener.php

```diff
@@ -0,0 +1,151 @@
+        private readonly TemplateVariables $globalVariables,
+        private readonly LegacyControllerContext $legacyControllerContext,
+        private readonly LanguageContext $languageContext,
+        private readonly ShopContext $shopContext,
+        private readonly CountryContext $countryContext,
+        private readonly LegacyContext $legacyContext,
+        private readonly MenuBuilder $menuBuilder,
+        private readonly Configuration $configuration,
+        private readonly string $imageDirectory,
+        private readonly string $picDirectory,
+        private readonly string $baseUri,
+    ) {
+    }
+
+    public function onKernelResponse(ResponseEvent $event): void
+    {
+        if (!$event->isMainRequest() || $this->isExternalApiRequest($event->getRequest())
+        ) {
+            return;
+        }
+
+        $this->legacyContext->getSmarty()->assign([
+            'maintenance_mode' => $this->globalVariables->isMaintenanceEnabled(),
+            'maintenance_allow_admins' => $this->globalVariables->isFrontOfficeAccessibleForAdmins(),
+            'debug_mode' => $this->globalVariables->isDebugMode(),
+            'breadcrumbs2' => 'TODO',
+            'help_link' => 'TODO',
+            'img_dir' => $this->imageDirectory,
+            'lite_display' => 'TODO',
+            'baseAdminUrl' => $this->baseUri . basename(_PS_ADMIN_DIR_) . '/',
+            'base_url' => $this->globalVariables->getBaseUrl(),
+            'toggle_navigation_url' => $this->legacyContext->getAdminLink('AdminEmployees', true, [
+                'action' => 'toggleMenu',
+            ]),
+            'collapse_menu' => $this->globalVariables->isMenuCollapsed(),
+            'enableSidebar' => 'TODO',
+            'title' => 'TODO',
+            'meta_title' => 'TODO',
+            'table' => 'TODO',
+            'tabs' => 'TODO',
+            'default_tab_link' => $this->globalVariables->getDefaultTabLink(),
+            'current_tab_level' => $this->menuBuilder->getCurrentTabLevel(),
+            'lang_is_rtl' => $this->languageContext->isRTL(),
+            'full_language_code' => $this->languageContext->getLanguageCode(),
+            'country_iso_code' => $this->countryContext->getIsoCode(),
+            'currentIndex' => $this->legacyControllerContext->currentIndex,
+            'default_language' => (int) $this->configuration->get('PS_LANG_DEFAULT'),
+            'js_router_metadata' => $this->globalVariables->getJsRouterMetadata(),
+            'locked_to_all_shop_context' => $this->globalVariables->isLockedToAllShopContext(),
+            'token' => $this->legacyControllerContext->token,
+            'employee' => $this->legacyContext->getContext()->employee,
+            'is_multishop' => $this->globalVariables->isMultiShop(),
+            'shop_name' => $this->shopContext->getName(),
+            'shop' => $this->legacyContext->getContext()->shop,
+            'shop_group' => $this->legacyContext->getContext()->shop->getGroup(),
```

Comment:
Add `@todo: This should be replaced with immutable ShopContext, but it means it needs a getGroup method that also returns an immutable ShopGroup DTO`